File: AnalizadorBigO/src/main/java/com/mycompany/analizadorbigo/BigOChart.java
package com.mycompany.analizadorbigo;

import org.jfree.chart.ChartFactory;
import org.jfree.chart.ChartPanel;
import org.jfree.chart.JFreeChart;
import org.jfree.data.xy.XYSeries;
import org.jfree.data.xy.XYSeriesCollection;

import javax.swing.*;
import java.util.Map;

public class BigOChart {

    public JPanel createChartPanel(Map<String, Integer> complexities) {
        // Crear dataset basado en las complejidades detectadas
        XYSeriesCollection dataset = createDataset(complexities);

        // Crear gráfico
        JFreeChart chart = ChartFactory.createXYLineChart(
                "Simulación de Complejidad Big O",
                "Tamaño de Entrada (n)",
                "Operaciones",
                dataset
        );

        // Configurar el panel del gráfico
        return new ChartPanel(chart);
    }

    private XYSeriesCollection createDataset(Map<String, Integer> complexities) {
        XYSeriesCollection dataset = new XYSeriesCollection();

        // Datos para O(1)
        if (complexities.containsKey("O(1)")) {
            XYSeries constant = new XYSeries("O(1)");
            for (int n = 1; n <= 10; n++) {
                constant.add(n, 1);
            }
            dataset.addSeries(constant);
        }

        // Datos para O(n)
        if (complexities.containsKey("O(n)")) {
            XYSeries linear = new XYSeries("O(n)");
            for (int n = 1; n <= 10; n++) {
                linear.add(n, n);
            }
            dataset.addSeries(linear);
        }

        // Datos para O(n^2)
        if (complexities.containsKey("O(n^2)")) {
            XYSeries quadratic = new XYSeries("O(n^2)");
            for (int n = 1; n <= 10; n++) {
                quadratic.add(n, n * n);
            }
            dataset.addSeries(quadratic);
        }

        // Datos para O(n^3)
        if (complexities.containsKey("O(n^3)")) {
            XYSeries cubic = new XYSeries("O(n^3)");
            for (int n = 1; n <= 10; n++) {
                cubic.add(n, n * n * n);
            }
            dataset.addSeries(cubic);
        }

        // Datos para O(n^4)
        if (complexities.containsKey("O(n^4)")) {
            XYSeries quartic = new XYSeries("O(n^4)");
            for (int n = 1; n <= 10; n++) {
                quartic.add(n, Math.pow(n, 4));
            }
            dataset.addSeries(quartic);
        }

        // Datos para O(log n)
        if (complexities.containsKey("O(log n)")) {
            XYSeries logarithmic = new XYSeries("O(log n)");
            for (int n = 1; n <= 10; n++) {
                logarithmic.add(n, Math.log(n));
            }
            dataset.addSeries(logarithmic);
        }

        // Datos para O(2^n)
        if (complexities.containsKey("O(2^n)")) {
            XYSeries exponential = new XYSeries("O(2^n)");
            for (int n = 1; n <= 10; n++) {
                exponential.add(n, Math.pow(2, n));
            }
            dataset.addSeries(exponential);
        }

        return dataset;
    }
}

File: pseudo/BigOChart.js
CLASE BigOChart

    MÉTODO createChartPanel(complexities: Map<String, Integer>) -> JPanel
        CREAR dataset USANDO createDataset(complexities)
        CREAR gráfico USANDO ChartFactory.createXYLineChart CON:
            - Título: "Simulación de Complejidad Big O"
            - Eje X: "Tamaño de Entrada (n)"
            - Eje Y: "Operaciones"
            - Dataset creado previamente
        RETORNAR nuevo ChartPanel USANDO el gráfico
    FIN MÉTODO

    MÉTODO PRIVADO createDataset(complexities: Map<String, Integer>) -> XYSeriesCollection
        CREAR dataset COMO nuevo XYSeriesCollection

        SI complexities CONTIENE "O(1)"
            CREAR serie constant COMO nuevo XYSeries CON nombre "O(1)"
            PARA n DESDE 1 HASTA 10
                AGREGAR punto (n, 1) A constant
            FIN PARA
            AGREGAR constant AL dataset
        FIN SI

        SI complexities CONTIENE "O(n)"
            CREAR serie linear COMO nuevo XYSeries CON nombre "O(n)"
            PARA n DESDE 1 HASTA 10
                AGREGAR punto (n, n) A linear
            FIN PARA
            AGREGAR linear AL dataset
        FIN SI

        SI complexities CONTIENE "O(n^2)"
            CREAR serie quadratic COMO nuevo XYSeries CON nombre "O(n^2)"
            PARA n DESDE 1 HASTA 10
                AGREGAR punto (n, n^2) A quadratic
            FIN PARA
            AGREGAR quadratic AL dataset
        FIN SI

        SI complexities CONTIENE "O(n^3)"
            CREAR serie cubic COMO nuevo XYSeries CON nombre "O(n^3)"
            PARA n DESDE 1 HASTA 10
                AGREGAR punto (n, n^3) A cubic
            FIN PARA
            AGREGAR cubic AL dataset
        FIN SI

        SI complexities CONTIENE "O(n^4)"
            CREAR serie quartic COMO nuevo XYSeries CON nombre "O(n^4)"
            PARA n DESDE 1 HASTA 10
                AGREGAR punto (n, n^4) A quartic
            FIN PARA
            AGREGAR quartic AL dataset
        FIN SI

        SI complexities CONTIENE "O(log n)"
            CREAR serie logarithmic COMO nuevo XYSeries CON nombre "O(log n)"
            PARA n DESDE 1 HASTA 10
                AGREGAR punto (n, log(n)) A logarithmic
            FIN PARA
            AGREGAR logarithmic AL dataset
        FIN SI

        SI complexities CONTIENE "O(2^n)"
            CREAR serie exponential COMO nuevo XYSeries CON nombre "O(2^n)"
            PARA n DESDE 1 HASTA 10
                AGREGAR punto (n, 2^n) A exponential
            FIN PARA
            AGREGAR exponential AL dataset
        FIN SI

        RETORNAR dataset
    FIN MÉTODO

FIN CLASE
